autoscheduler/frontend/src/components/Navbar/LoginButton.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { Button } from '@material-ui/core';
+import ExitToAppOutlinedIcon from '@material-ui/icons/ExitToAppOutlined';
+// This file contains the login button. When a user logs in, it is replaced by their name
+// followed by the logout button.
+const LoginButton: React.FC = () => {
+  // Used to track whether the API call has returned anything yet
+  const [loading, setLoading] = React.useState(true);
+
+  // used to logout the user
+  function logout(): void {
+    fetch('sessions/logout').then(
+      (res) => {
+        if (res.ok) {
+          window.location.reload();
+          return;
+        }
+        throw new Error(res.status.toString());
```

Comment:
Why throw an error if you're going to catch it without handling it 2 lines later? Is that for future updates maybe?

Reply:
The catch is meant to stop the website from crashing if the API gives a bad response. It was in the getName UseEffect because when the response didn't have JSON it caused the website to crash. I made this one by copy pasting the getName one and then modifying it. Since I'm not actually processing any of the response other than checking `if(res.ok)` I can probably get rid of it here without any consequences.